cirq-core/cirq/experiments/two_qubit_xeb.py

```diff
@@ -156,9 +157,152 @@ def plot_histogram(self, ax: Optional[plt.Axes] = None, **plot_kwargs) -> plt.Ax
+        }
+
+    @cached_method
+    def inferred_xeb_error(self) -> Dict[Tuple['cirq.GridQubit', 'cirq.GridQubit'], float]:
+        """Return the inferred XEB error for all pairs."""
+        return {
+            pair: 1 - noise_utils.decay_constant_to_xeb_fidelity(decay, 2)
+            for pair, decay in self.inferred_decay_constant().items()
+        }
+
+    def _target_errors(self, target_error: str) -> Dict[Tuple['cirq.GridQubit', ...], float]:
+        error_funcs = {
+            'pauli': self.inferred_pauli_error,
+            'decay_constant': self.inferred_decay_constant,
+            'xeb': self.inferred_xeb_error,
+        }
+        return cast(Dict[Tuple['cirq.GridQubit', ...], float], error_funcs[target_error]())
+
+    def plot_heatmap(
+        self, target_error: str = 'pauli', ax: Optional[plt.Axes] = None, **plot_kwargs
+    ) -> plt.Axes:
+        """plot the heatmap of the target errors.
+
+        Args:
+            target_error: The error to draw. Must be one of 'xeb', 'pauli', or 'decay_constant'
+            ax: the plt.Axes to plot on. If not given, a new figure is created,
+                plotted on, and shown.
+            **plot_kwargs: Arguments to be passed to 'plt.Axes.plot'.
+        """
+        show_plot = not ax
+        if not isinstance(ax, plt.Axes):
+            fig, ax = plt.subplots(1, 1, figsize=(8, 8))
+        heatmap_data = self._target_errors(target_error)
+
+        name = f'{target_error} error' if target_error != 'decay_constant' else 'decay constant'
+        ax.set_title(f'device {name} heatmap')
+
+        vis.TwoQubitInteractionHeatmap(heatmap_data).plot(ax=ax, **plot_kwargs)
+        if show_plot:
+            fig.show()
+        return ax
+
+    def plot_histogram(
+        self,
+        target_error: str = 'pauli',
+        ax: Optional[plt.Axes] = None,
+        single_qubit: bool = True,
+        two_qubit: bool = True,
+        **plot_kwargs,
+    ) -> plt.Axes:
+        """plot a histogram of target error.
+
+        Args:
+            target_error: The error to draw. Must be one of 'xeb', 'pauli', or 'decay_constant'
+            single_qubit: Whether to plot the single-qubit RB pauli errors.
+            two_qubit: Whether to plot the two-qubit inferred pauli errors.
+            ax: the plt.Axes to plot on. If not given, a new figure is created,
+                plotted on, and shown.
+            **plot_kwargs: Arguments to be passed to 'plt.Axes.plot'.
+
+        Raises:
+            ValueError: If
+                - single_qubit and two_qubit are both False.
+                - single_qubit is True and target_error is not 'pauli'.
+        """
+        if not single_qubit and not two_qubit:
+            raise ValueError("At least one of single_qubit and two_qubit must be True")
+        if single_qubit and target_error != 'pauli':
+            raise ValueError("single_qubit must be False if target_error is not 'pauli'")
```

Comment:
This is a bit clunky API - if users want to show xeb histogram they need to pass 2 arguments `plot_histogram('xeb', single_qubit=False)`.  

How about replacing `single_qubit`, `two_qubit` arguments with an argument `kind = one of {"combined", "single_qubit", "two_qubit"}` where "combined" would be the default and would plot whatever makes sense for the target_error, and `single_qubit`, `two_qubit` would plot just one kind of errors.

Reply:
done